src/components/DataTable.tsx

```diff
@@ -233,6 +238,16 @@ export function DataTable<T extends object>(props: Props<T>) {
       });
   };
 
+  useEffect(() => {
+    if (props.includeObjectSelection && data) {
+      const filteredSelectedResources = data.data.data.filter((resource: any) =>
```

Comment:
@beganovich Hmm, I'm pretty sure there is no way to avoid using "any" as a type. If we define it as "T", where "T" extends the object, it will not allow us to use "resource.id" (it will say something like "we do not have the 'id' property on the type object"). If we try to make some adjustments like "resource['id' as keyof typeof resource]", it will not pass other checkers, such as Vite when pushing to GitHub, I guess. Another approach is "resource['id' as keyof T]", but that also breaks when pushing to GitHub.

So, I'm not sure if there is a way to avoid it in this specific case. But let me know your thoughts.